File: Leetcode 3622 Check Divisibility by Digit Sum and Product.py
# 2026/08/22
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        dig_sum = 0
        dig_pro = 1
        for i in str(n):
            dig_sum += int(i)
            dig_pro *= int(i)
        return n % (dig_sum+dig_pro) == 0
            

File: Leetcode 3622 Check Divisibility by Digit Sum and Product.go
// 2026/08/22
func checkDivisibility(n int) bool {
    dig_sum := 0
    dig_pro := 1
    for _, ch := range strconv.Itoa(n) {
        a := int(ch - '0')
        dig_sum += a
        dig_pro *= a
    }
    return n % (dig_sum+dig_pro) == 0
}
